### PR TITLE
Added NuclearCraft, NuclearCraft: Overhauled and Tech Reborn Methods

### PR DESCRIPTION
**Pull request by Redfire75369**
Nuclearcraft Overhaul will be made soon due to slight differences in the methods

### CODE REVIEW

**Comment by jaredlll08** [comment]
everywhere you have \`\`\`kotlin, I'm going to need you to change that to \`\`\`zenscript

Both github and our docs have support for it

**Comment by Redfire75369** [comment]
Oh, ok, back when I did this, that didn't exist

File: docs/Mods/NuclearCraft/Machine Recipes/Fluid_Enricher.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Fluid Enricher
+
+## Importing the Package
+`mods.nuclearcraft.dissolver`
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft/Machine Recipes/Fluid_Enricher.md]: It is actually called Dissolver

File: docs/Mods/NuclearCraft/Machine Recipes/Condenser.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Condense
+
+## Importing the Package
+`mods.nuclearcraft.condenser`
+
+## Adding Recipes
+```kotlin
+mods.nuclearcraft.condenser.addRecipe([fluidInput, fluidOutput, @Optional double coolingRequired, @Optional int condensingTemperature}]);
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft/Machine Recipes/Condenser.md]: Fixed in most rexent commit.

File: docs/Mods/NuclearCraft/Machine Recipes/Decay_Generator.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Decay Generator
+
+## Importing the Package
+`mods.nuclearcraft.decay_generator`
+
+## Adding Recipes
+```kotlin
+mods.nuclearcraft.decay_generator.addRecipe([blockInput, blockOutput, double lifetimeTicks, double energyPerSecond, @Optional double processRadiation]);
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft/Machine Recipes/Decay_Generator.md]: Improved upon in most recent commit. Check Important_Info.md

File: docs/Mods/NuclearCraft/Machine Recipes/Supercooler.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Supercooler
+
+## Importing the Package
+`mods.nuclearcraft.decay_hastener`
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft/Machine Recipes/Supercooler.md]: Fixed in most recent commit

**Comment by Redfire75369** [comment]
The Mods, NuclearCraft: Overhauled and Tech Reborn have had their documentation added as well.

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Condense
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md:1]: Condense**r**?

Or were you trying to use the verb here, in conflict to everywhere else 😛

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+**Note: Partially Broken**
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md:2]: In what sense?

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+## Importing the Package
+`mods.nuclearcraft.Condenser`
+
+## Adding Recipes
+```zenscript
+mods.nuclearcraft.Condenser.addRecipe(ILiquidStack fluidInput, ILiquidStack fluidOutput, @Optional double coolingRequired, @Optional int condensingTemperature});
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md:9]: What's the `}` for 😛

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Heat_Exchanger.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Heat Exchanger
+**Note: Partially Broken**
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Heat_Exchanger.md:2]: In what way?

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Salt Fission
+**Note: Partially Broken**
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md:2]: In what way?

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+## Importing the Package
+`mods.nuclearcraft.SaltFission`
+
+## Adding Recipes
+```zenscript
+mods.nuclearcraft.SaltFission.addRecipe(fluidInput, fluidOutput, double baseTime, double basePower, @Optional double processRadiation);
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md:9]: The types for FluidIn/Out are missing

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+```
+
+## Removing Recipes
+```zenscript
+mods.nuclearcraft.SaltFission.removeRecipeWithInput(fluidInput);
+mods.nuclearcraft.SaltFission.removeRecipeWithOutput(fluidOutput);
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Fission.md:15]: The types for fluidInput/Output are missing

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Mixer.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# Salt Mixer
+
+## Importing the Package
+`mods.nuclearcraft.salt_mixer`
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Mixer.md:4]: It's SaltMixer here.
You switched it with the non-overhauled version

File: docs/Mods/NuclearCraft_Overhauled/Radiation.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+IEntityLivingBase::getRadiation();
+```
+
+### Radaway Buffer
+`amount` is the amount of Radaway Buffer added to the `IEntityLivingBase`. 
+`slowBuffer` decides if the Radaway Buffer added is a slow buffer.
+```zenscript
+IEntityLivingBase::addRadawayBuffer(double amount, @Optional boolean slowBuffer);
+IEntityLivingBase::setRadawayBuffer(double amount, @Optional boolean slowBuffer);
+IEntityLivingBase::getRadawayBuffer({boolean slowBuffer});
```
Comment (kindlich) [inline docs/Mods/NuclearCraft_Overhauled/Radiation.md:71]: You're breaking your own style here, did you mean to use an `@Optional` here?

File: docs/Mods/Tech_Reborn/Alloy_Smelter.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+`mods.techreborn.alloySmelter`
+
+## Adding Recipes
+```zenscript
+mods.techreborn.alloySmelter.addRecipe(IItemStack output, IIngredient input1, IIngredient input2, int ticktime, int euTick);
+```
+
+## Removing Recipes
+```zenscript
+mods.techreborn.alloySmelter.removeInputRecipe(IIngredient iIngredient);
```
Comment (kindlich) [inline docs/Mods/Tech_Reborn/Alloy_Smelter.md:13]: `IItemStack ingredientA, IItemStack ingredientB`

File: docs/Mods/Tech_Reborn/Assembling_Machine.md [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+`mods.techreborn.assemblingMachine`
+
+## Adding Recipes
+```zenscript
+mods.techreborn.assemblingMachine.addRecipe(IItemStack output, IIngredient input1, IIngredient input2, int ticktime, int euTick);
+```
+
+## Removing Recipes
+```zenscript
+mods.techreborn.assemblingMachine.removeInputRecipe(IIngredient iIngredient);
```
Comment (kindlich) [inline docs/Mods/Tech_Reborn/Assembling_Machine.md:13]: IItemStack input1, IItemStack input2

File: docs/Mods/Tech_Reborn/Chemical_Reactor.md [hunk 16]
```diff
@@ -0,0 +1,16 @@
+# Chemical Reactor
+
+## Importing the Package
+`mods.techreborn.chemicalReactorRecipe`
```
Comment (kindlich) [inline docs/Mods/Tech_Reborn/Chemical_Reactor.md:4]:
```suggestion
`mods.techreborn.chemicalReactor`
```

**Comment by Redfire75369** [comment]
Oh, I forgot those 3 Machines because they aren't directly on https://wiki.techreborn.ovh/doku.php?id=development:mt.

File: docs/Mods/NuclearCraft/Machine_Recipes/Salt_Mixer.md [hunk 17]
```diff
@@ -0,0 +1,16 @@
+# Salt Mixer
+
+## Importing the Package
+`mods.nuclearcraft.SaltMixer`
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft/Machine_Recipes/Salt_Mixer.md]: Fixed in most recent commit

File: docs/Mods/NuclearCraft_Overhauled/Important_Info.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+The size of the stack produced on each process is randomly assigned from a binomial distribution specified by the chance
+info. The percentage and ingredient stack size play the roles of the probability and number of trials, respectively. A
+minimum stack size can also be specified - without this, the minimum stack size is simply 0.
+
+For ChanceFluidIngredients, a 'stack difference' must also be specified, which determines the difference in size between
+the possible stacks (for ChanceItemIngredients, this is effectively 1). For example, a ChanceFluidIngredient for an
+ingredient of size 500, with a stack difference of 150 and minimum stack size of 50, will produce 50, 200, 350 or 500
+millibuckets of the fluid.
+
+### ChanceItemIngredient
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft_Overhauled/Important_Info.md:28]: Improved in most recent commit

File: docs/Mods/NuclearCraft_Overhauled/Important_Info.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+
+#### Extra Methods
+
+```zenscript
+IIngredient getInternalIngredient();
+int getChancePercent();
+int getMinStackSize();
+```
+
+### ChanceFluidIngredient
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply docs/Mods/NuclearCraft_Overhauled/Important_Info.md:51]: Improved in most recent commit

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md (hunk 5 above)
Comment (Redfire75369) [reply docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Condenser.md]: Fixed in most recent commit

File: docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Mixer.md (hunk 12 above)
Comment (Redfire75369) [reply docs/Mods/NuclearCraft_Overhauled/Machine_Recipes/Salt_Mixer.md:4]: Fixed in most recent commit

File: docs/Mods/NuclearCraft_Overhauled/Radiation.md (hunk 13 above)
Comment (Redfire75369) [reply docs/Mods/NuclearCraft_Overhauled/Radiation.md]: Added Explanation in latest commit

File: docs/Mods/Tech_Reborn/Alloy_Smelter.md (hunk 14 above)
Comment (Redfire75369) [reply docs/Mods/Tech_Reborn/Alloy_Smelter.md]: Fixed in most recent commit

File: docs/Mods/Tech_Reborn/Assembling_Machine.md (hunk 15 above)
Comment (Redfire75369) [reply docs/Mods/Tech_Reborn/Assembling_Machine.md]: Fixed in most recent commit

File: docs/Mods/Tech_Reborn/Chemical_Reactor.md (hunk 16 above)
Comment (Redfire75369) [reply docs/Mods/Tech_Reborn/Chemical_Reactor.md]: Fixed in most recent commit

**Pull request merged**